tests/checkpoint/test_v1_checkpoint.py

```diff
@@ -367,6 +368,37 @@ def test_checkpoint_deserialization_failure(
 
         assert expected_error in str(e.value)
 
+    @pytest.mark.unit
+    def test_checkpoint_deserialization_with_actions(self, mocker: MockerFixture):
+        # Arrange
+        context = mocker.Mock(spec=AbstractDataContext)
+        context.validation_definition_store.get.return_value = mocker.Mock(
+            spec=ValidationDefinition
+        )
+        set_context(context)
+
+        # Act
+        serialized_checkpoint = {
+            "actions": [
+                {"site_names": [], "type": "update_data_docs"},
+                {"slack_webhook": "test", "type": "slack"},
+                {"teams_webhook": "test", "type": "microsoft"},
+            ],
+            "id": "e7d1f462-821b-429c-8086-cca80eeea5e9",
+            "name": "my_checkpoint",
+            "result_format": "SUMMARY",
+            "validation_definitions": [
+                {"id": "3fb9ce09-a8fb-44d6-8abd-7d699443f6a1", "name": "my_validation_def"}
+            ],
+        }
+        checkpoint = Checkpoint.parse_obj(serialized_checkpoint)
+
+        # Assert
+        assert len(checkpoint.actions) == 3
+        assert isinstance(checkpoint.actions[0], UpdateDataDocsAction)
+        assert isinstance(checkpoint.actions[1], SlackNotificationAction)
+        assert isinstance(checkpoint.actions[2], MicrosoftTeamsNotificationAction)
```

Comment:
I'd probably add another test here to verify that (pseudocode) serialized_checkpoint == serialize(deserialize(checkpoint))